### PR TITLE
Pr/openthread command line

### PR DESCRIPTION
**Pull request by MarekPorwisz**
Added command line sample + doc

Since thread examples do not compile on current nordic master, creating internal pr so you could start reviewing internally.

### CODE REVIEW

File: samples/net/openthread/ot_cli/src/main.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#include <zephyr.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(cli_sample, CONFIG_OT_COMMAND_LINE_INTERFACE_LOG_LEVEL);
+
+
+void main(void)
+{
+	LOG_INF("OpenThread Command Line Interface is now running.");
+	LOG_INF("Use the 'ot' keyword to invoke OpenThread commands e.g. '%s.'",
```
Comment (lmaciejonczyk) [inline samples/net/openthread/ot_cli/src/main.c]: Unnecessary use of `%s` to incorporate static string into log message.

File: samples/net/openthread/ot_cli/src/main.c [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+LOG_MODULE_REGISTER(cli_sample, CONFIG_OT_COMMAND_LINE_INTERFACE_LOG_LEVEL);
+
+
+void main(void)
+{
+	LOG_INF("OpenThread Command Line Interface is now running.");
+	LOG_INF("Use the 'ot' keyword to invoke OpenThread commands e.g. "
+		"'ot thread start.'");
+	LOG_INF("For the full commands list refer to the OpenThread CLI");
+	LOG_INF("documentation at:");
```
Comment (edmont) [inline samples/net/openthread/ot_cli/src/main.c]: Append this line as a continuation of the previous one? (inline)

**Review by edmont — APPROVED** [review]
LGTM :+1:

File: samples/net/openthread/ot_cli/src/main.c [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	LOG_INF("https://github.com/openthread/openthread/blob/master/src/cli/README.md");
```
Comment (lmaciejonczyk) [inline samples/net/openthread/ot_cli/src/main.c]: Why not use only one LOG_INF for this?

**Review by greg-fer — APPROVED** [review]
Added my commit with edits. Please verify it before merge.

File: samples/net/openthread/ot_cli/src/main.c (hunk 3 above)
Comment (MarekPorwisz) [reply samples/net/openthread/ot_cli/src/main.c]: changed as requested. Added some extra newlines to make it more visible:

uart:~$ *** Booting Zephyr OS build v2.1.99-ncs1-1145-g13dc96f83bab  ***


[00:00:00.467,498] <inf> net_l2_openthread: State changed! Flags: 0x00000010 Current role: 1
[00:00:00.467,529] <inf> net_l2_openthread: State changed! Flags: 0x00000020 Current role: 1
[00:00:00.681,457] <inf> net_config: Initializing network
[00:00:00.806,457] <inf> net_l2_openthread: State changed! Flags: 0x000010e4 Current role: 3
[00:00:00.822,631] <inf> net_l2_openthread: State changed! Flags: 0x00000200 Current role: 3
[00:00:01.581,695] <inf> cli_sample:

OpenThread Command Line Interface is now running.
Use the 'ot' keyword to invoke OpenThread commands e.g. 'ot thread start.'
For the full commands list refer to the OpenThread CLIdocumentation at:
https://github.com/openthread/openthread/blob/master/src/cli/README.md

uart:~$

**Review by lmaciejonczyk — APPROVED** [review]
It looks ok to me

File: samples/net/openthread/ot_cli/prj.conf [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# Generic networking options
+CONFIG_NETWORKING=y
+CONFIG_NET_UDP=y
+CONFIG_NET_IPV6=y
+CONFIG_NET_SOCKETS=y
```
Comment (rlubos) [inline samples/net/openthread/ot_cli/prj.conf]: Do we really need sockets here? They're not used by the sample (there might be some hidden dependencies I'm not aware of though).

File: samples/net/openthread/ot_cli/prj.conf [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+# Network shell
+CONFIG_NET_SHELL=y
+
+CONFIG_CPLUSPLUS=y
+
+CONFIG_NET_CONFIG_SETTINGS=y
+CONFIG_NET_IPV6_NBR_CACHE=n
+CONFIG_NET_IPV6_MLD=n
+CONFIG_NET_CONFIG_NEED_IPV6=y
+CONFIG_NET_CONFIG_INIT_TIMEOUT=3
```
Comment (rlubos) [inline samples/net/openthread/ot_cli/prj.conf]: Couldn't this be 0 for the CLI app? It's not taking any action on its own after startup.

File: samples/net/openthread/ot_cli/prj.conf [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+# Other OpenThread dependencies
+CONFIG_FLASH=y
+CONFIG_FLASH_PAGE_LAYOUT=y
+CONFIG_FLASH_MAP=y
+CONFIG_MPU_ALLOW_FLASH_WRITE=y
+CONFIG_NVS=y
+CONFIG_SETTINGS=y
+
+CONFIG_REBOOT=y
+
```
Comment (rlubos) [inline samples/net/openthread/ot_cli/prj.conf]: Double empty line at the end of the file.

File: samples/net/openthread/ot_cli/src/main.c [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+LOG_MODULE_REGISTER(cli_sample, CONFIG_OT_COMMAND_LINE_INTERFACE_LOG_LEVEL);
+
+#define WELLCOME_TEXT \
+	"\n\r"\
+	"\n\r"\
+	"OpenThread Command Line Interface is now running.\n\r" \
+	"Use the 'ot' keyword to invoke OpenThread commands e.g. " \
+	"'ot thread start.'\n\r" \
+	"For the full commands list refer to the OpenThread CLI" \
```
Comment (edmont) [inline samples/net/openthread/ot_cli/src/main.c]:
```suggestion
	"For the full commands list refer to the OpenThread CLI " \
```

**Pull request closed without merging**